cirq-ft/cirq_ft/algos/arithmetic_gates.py

```diff
@@ -130,6 +130,67 @@ def _t_complexity_(self) -> infra.TComplexity:
         )
 
 
+def mix_double_qubit_registers(
+    context: cirq.DecompositionContext,
+    msb: Tuple[cirq.Qid, cirq.Qid],
+    lsb: Tuple[cirq.Qid, cirq.Qid],
+) -> cirq.OP_TREE:
+    """Generates the COMPARE2 circuit from https://www.nature.com/articles/s41534-018-0071-5#Sec8"""
+    [ancilla] = context.qubit_manager.qalloc(1)
+    x_1, y_1 = msb
+    x_0, y_0 = lsb
+
+    def _cswap(c: cirq.Qid, a: cirq.Qid, b: cirq.Qid) -> cirq.OP_TREE:
+        [q] = context.qubit_manager.qalloc(1)
+        yield cirq.CNOT(a, b)
+        yield and_gate.And().on(c, b, q)
+        yield cirq.CNOT(q, a)
+        yield cirq.CNOT(a, b)
+
+    yield cirq.X(ancilla)
+    yield cirq.CNOT(y_1, x_1)
+    yield cirq.CNOT(y_0, x_0)
+    yield _cswap(x_1, x_0, ancilla)
+    yield _cswap(x_1, y_1, y_0)
+    yield cirq.CNOT(y_0, x_0)
+
+
+def compare_qubits(
+    x: cirq.Qid, y: cirq.Qid, less_than: cirq.Qid, greater_than: cirq.Qid
+) -> cirq.OP_TREE:
+    """Generates the comparison circuit from https://www.nature.com/articles/s41534-018-0071-5#Sec8
+
+    Args:
+        x: first qubit of the comparison and stays the same after circuit execution.
+        y: second qubit of the comparison.
+            This qubit will store equality value `x==y` after circuit execution.
+        less_than: Assumed to be in zero state. Will store `x < y`.
+        greater_than: Assumed to be in zero state. Will store `x > y`.
+
+    Returns:
+        The circuit in (Fig. 3) in https://www.nature.com/articles/s41534-018-0071-5#Sec8
+    """
+
+    yield and_gate.And([0, 1]).on(x, y, less_than)
+    yield cirq.CNOT(less_than, greater_than)
+    yield cirq.CNOT(y, greater_than)
+    yield cirq.CNOT(x, y)
+    yield cirq.CNOT(x, greater_than)
+    yield cirq.X(y)
+
+
+def _equality_with_zero(
+    context: cirq.DecompositionContext, X: Tuple[cirq.Qid, ...], z: cirq.Qid
+) -> cirq.OP_TREE:
```

Comment:
little docstring please!

can we use pep8 names instead of `X`? 

Suggest loosening input type annotation to `Sequence[cirq.Qid]`.

Reply:
done